Detectors/Upgrades/ALICE3/Passive/src/Pipe.cxx

```diff
@@ -93,75 +94,140 @@ void Alice3Pipe::ConstructGeometry()
+      if (iLayer != 0) {
+        subtractorsFormula += Form("+");
       }
+      subtractorsFormula += Form("TRKLAYER_%dsh", iLayer);
     }
 
-    // Escavate vacuum for hosting cold plate
-    TGeoTube* coldPlate = new TGeoTube("TRK_COLDPLATEsh", 2.6f, 2.6f + 150.e-3, 50.f / 2);
+    // IRIS vacuum vessel and coldplate dimensions
+    float coldplateRIn = 2.6f;              // cm
+    float coldplateThick = 150.e-3;         // cm
+    float coldplateLength = 50.f;           // cm
+    float irisVacuumVesselInnerRIn = 0.48f; // cm
+    float irisVacuumVesselOuterRIn = coldplateRIn + coldplateThick;
+    float irisVacuumVesselLength = 70.f;   // cm
+    float irisVacuumVesselThick = 150.e-4; // cm
+
+    // Excavate vacuum for hosting cold plate and IRIS tracker
+    TGeoTube* coldPlate = new TGeoTube("TRK_COLDPLATEsh", coldplateRIn, coldplateRIn + coldplateThick, coldplateLength / 2.);
     subtractorsFormula += "+TRK_COLDPLATEsh";
 
+    TGeoTube* irisVacuumVesselInner = new TGeoTube("TRK_IRISVACUUMVESSELINNERsh", irisVacuumVesselInnerRIn, irisVacuumVesselInnerRIn + irisVacuumVesselThick, irisVacuumVesselLength / 2.);
+    subtractorsFormula += "+TRK_IRISVACUUMVESSELINNERsh";
+
+    TGeoTube* irisVacuumVesselOuter = new TGeoTube("TRK_IRISVACUUMVESSELOUTERsh", irisVacuumVesselOuterRIn, irisVacuumVesselOuterRIn + irisVacuumVesselThick, irisVacuumVesselLength / 2.);
+    subtractorsFormula += "+TRK_IRISVACUUMVESSELOUTERsh";
+
+    TGeoTube* irisVacuumVesselWall = new TGeoTube("TRK_IRISVACUUMVESSELWALLsh", irisVacuumVesselInnerRIn, irisVacuumVesselOuterRIn + irisVacuumVesselThick, irisVacuumVesselThick / 2.);
+    TGeoTranslation* posIrisVacVWallNegZSide = new TGeoTranslation("IRISWALLNEGZ", 0., 0., -irisVacuumVesselLength / 2. - irisVacuumVesselThick / 2.);
+    posIrisVacVWallNegZSide->RegisterYourself();
+    subtractorsFormula += "+TRK_IRISVACUUMVESSELWALLsh:IRISWALLNEGZ";
+
+    TGeoTranslation* posIrisVacVWallPosZSide = new TGeoTranslation("IRISWALLPOSZ", 0., 0., irisVacuumVesselLength / 2. + irisVacuumVesselThick / 2.);
+    posIrisVacVWallPosZSide->RegisterYourself();
+    subtractorsFormula += "+TRK_IRISVACUUMVESSELWALLsh:IRISWALLPOSZ";
+  }
+
+  if (!mIsFT3Activated) {
+    std::vector<TGeoTube*> ft3DiscShapes;
+    std::vector<TGeoTranslation*> ft3DiscPositions;
+
+    std::vector<std::array<float, 4>> discsQuotas = {std::array<float, 4>{0.5f, 2.5f, 100.e-4, 26.}, // TODO: Set discs dynamically. {rIn, rOut, thickness, zpos}
+                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-4, 30.},
+                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-4, 34.},
+                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-4, -26.},
+                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-4, -30.},
+                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-4, -34.}};
```

Comment:
From FT3 Detector definition:
```c++
float layersx2X0 = 1.e-2;
std::vector<std::array<float, 5>> layersConfig{
{26., .5, 2.5, 0.1f * layersx2X0}, // {z_layer, r_in, r_out, Layerx2X0},
...
}
```
So I guess it is:
```suggestion
    std::vector<std::array<float, 4>> discsQuotas = {std::array<float, 4>{0.5f, 2.5f, 100.e-3, 26.}, // TODO: Set discs dynamically. {rIn, rOut, thickness, zpos}
                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-3, 30.},
                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-3, 34.},
                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-3, -26.},
                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-3, -30.},
                                                     std::array<float, 4>{0.5f, 2.5f, 100.e-3, -34.}};
```

Reply:
Good catch, but I believe we are both wrong.
In this line
https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/Upgrades/ALICE3/FT3/simulation/src/FT3Layer.cxx#L55
The thickness of a FT3 layer is set based on the x/x0 you give it. 
So it should actually be 95.e-4 instead of 100.e-3 or 100.e-4

Now that I think about it, if you supply the geometry for the FT3 from a configuration file, then pipe.cxx will not know which volumes to excavate